README.dev.md

```diff
@@ -6,11 +6,612 @@ You will need a machine with at least 2GB of RAM available (see
+    https://landscape.canonical.com/
+
+  Get cloud support with Ubuntu Advantage Cloud Guest:
+    http://www.ubuntu.com/business/services/cloud
+
+0 packages can be updated.
+0 updates are security updates.
+```
+
+Or something as brief as this in the case of Ubuntu:
+
+```
+Welcome to Ubuntu 14.04.1 LTS (GNU/Linux 4.4.0-21-generic x86_64)
+
+ * Documentation:  https://help.ubuntu.com/
+```
+
+Congrats, you're now inside the Zulip dev environment! 
+
+You can confirm this by looking at the command prompt, which starts with
+`(zulip-venv)`. 
+
+Next you'll navigate to the shared Zulip directory `/srv/zulip`, activate the
+python venv, and start the web server:
+
+``` 
+(zulip-venv)vagrant@vagrant-ubuntu-trusty-64:~
+$ cd /srv/zulip 
+(zulip-venv)vagrant@vagrant-ubuntu-trusty-64:
+~$ source /srv/zulip-venv/bin/activate 
+(zulip-venv)vagrant@vagrant-ubuntu-trusty-64:
+~$ ./tools/run-dev.py --interface='' 
+```
+
+You will see several lines of output starting with something like:
+
+``` 
+2016-05-04 22:20:33,895 INFO: process_fts_updates starting
+Recompiling templates
+2016-05-04 18:20:34,804 INFO: Not in recovery; listening for FTS updates
+done
+Validating Django models.py...
+System check identified no issues (0 silenced).
+
+Django version 1.8
+Tornado server is running at http://localhost:9993/
+Quit the server with CTRL-C.
+2016-05-04 18:20:40,716 INFO     Tornado loaded 0 event queues in 0.001s
+2016-05-04 18:20:40,722 INFO     Tornado  95.5% busy over the past  0.0 seconds
+Performing system checks...
+``` 
+And ending with something similar to:
+
+``` 
+http://localhost:9994/webpack-dev-server/
+webpack result is served from http://localhost:9991/webpack/
+content is served from /srv/zulip
+
+webpack: bundle is now VALID.
+2016-05-06 21:43:29,553 INFO     Tornado  31.6% busy over the past 10.6 seconds
+2016-05-06 21:43:35,007 INFO     Tornado  23.9% busy over the past 16.0 seconds
+``` 
+
+Now the Zulip web server should be running and accessible. Verify this by
+navigating to [http://localhost:9991/](http://localhost:9991/) in your browser
+on your main machine.
+
+The web server will continue to run and send output to the terminal window.
+When you navigate to Zulip in your browser, check your terminal and you
+should see something like:
+
+``` 
+2016-05-04 18:21:57,547 INFO     127.0.0.1       GET     302 582ms (+start: 417ms) / (unauth via ?)
+[04/May/2016 18:21:57]"GET / HTTP/1.0" 302 0
+2016-05-04 18:21:57,568 INFO     127.0.0.1       GET     301   4ms /login (unauth via ?)
+[04/May/2016 18:21:57]"GET /login HTTP/1.0" 301 0
+2016-05-04 18:21:57,819 INFO     127.0.0.1       GET     200 209ms (db: 7ms/2q) /login/ (unauth via ?)
+``` 
+
+Now you're ready for [Step 4: Developing.](#step-4-developing)
+
+### Step 4: Developing
+
+#### Where to edit files
+
+You'll work by editing files on your main machine, in the directory where you
+cloned Zulip. Use your favorite editor (Sublime, Atom, Vim, Emacs, Notepad++,
+etc.)  
```

Comment:
Do you think this is a good place to briefly mention that Zulip uses 4-space indentation (no tabs) and so they may want to configure their editor to use that if that isn't the default?